src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -287,37 +330,54 @@ const StudentApplication = ({ userProfile, authentication }) => {
    *
    * @async
    * @function changeApplicationEditor
-   * @param {String} newEditorUsername the student username of the person who will be allowed to edit this application
+   * @param {StudentProfileInfo} newEditorProfile the StudentProfileInfo object for the person who will be allowed to edit this application
    */
-  const changeApplicationEditor = async (newEditorUsername) => {
+  const changeApplicationEditor = async (newEditorProfile) => {
     setSaving(true);
     setSaveButtonAlertTimeout(null);
     try {
       const result = await housing.changeApartmentAppEditor(
         applicationDetails.ApplicationID,
-        newEditorUsername,
+        newEditorProfile.AD_Username,
       );
-      console.log(result); //! DEBUG
-      setApplicationDetails((prevApplicationDetails) => ({
-        ...prevApplicationDetails,
-        EditorProfile: newEditorProfile,
-      }));
-      setSaving('success');
-      setUnsavedChanges(true);
-      // loadApplication(); //? Coming soon to a feature near you
-    } catch {
-      setSnackbarText('Something went wrong while trying to save the new application editor.');
-      setSnackbarSeverity('error');
-      setSnackbarOpen(true);
-      setSaving('failed');
+      if (result) {
+        try {
+          const result = loadApplication();
+          if (!result) {
+            throw new Error('Failed to load apartment application.');
+          }
+        } catch {
```

Comment:
```suggestion
        try {
          const loadingResult = loadApplication();
          if (!loadingResult ) {
            throw new Error('Failed to load apartment application.');
          }
        } catch {
```
https://github.com/gordon-cs/gordon-360-ui/pull/1093/files#r625092241